packages/nuxt/src/app/composables/preview.ts

```diff
@@ -0,0 +1,76 @@
+import { toRef, watch } from 'vue'
+
+import { useState } from './state'
+import { refreshNuxtData } from './asyncData'
+import { useRoute, useRouter } from './router'
+
+interface Preview {
+  enabled: boolean
+  state: Record<any, unknown>
+}
+
+type EnteredState = Record<any, unknown> | null | undefined | void
+
+let unregisterRefreshHook: (() => any) | undefined
+
```

Comment:
If this lands in 3.10.0, please add:
```suggestion

/** @since 3.10.0 */
```